unit_py/test_double_spring_mass.py

```diff
@@ -179,5 +179,122 @@ def external_force(t, q, v, f):
         # Compare the numerical and analytical solutions
         self.assertTrue(np.allclose(x_jiminy, x_analytical, atol=TOLERANCE))
 
+    def test_fixed_body_constraint(self):
+        '''
+        @brief Test kinematic constraint: fixed second mass with a constaint.
+        '''
+        # Set same spings as usual
+        def compute_command(t, q, v, sensor_data, u):
+            u[:] = 0.0
+
+        def internal_dynamics(t, q, v, sensor_data, u):
+            u[:] = - self.k * q - self.nu * v
+        controller = jiminy.ControllerFunctor(compute_command, internal_dynamics)
+        controller.initialize(self.robot)
+        engine = jiminy.Engine()
+        engine.initialize(self.robot, controller)
+
+        # Add a kinematic constraint on the second mass
+        constraint = jiminy.FixedFrameConstraint("SecondMass")
+        self.robot.add_constraint("fixMass", constraint)
+
+        # Run simulation
+        engine.simulate(self.tf, self.x0)
+
+        log_data, _ = engine.get_log()
+        time = log_data['Global.Time']
+        x_jiminy = np.stack([log_data['HighLevelController.' + s]
```

Comment:
In my opinion, '.'.join('HighLevelController', s) is more robust and clear.

Reply:
I find direct contatenation clearer than join - especially with '.' as joint word, it's easy to miss it and to misunderstand what the function is doing. Is join really more robust ?

Reply:
Yes it is, since the delimiter and telemetry names could easily be exposed in Python, and used to automatically generate the appropriate fieldname, no matter the recursion level.